components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -770,19 +817,26 @@ internal sealed class SubscriptionWrapper : ISubscription {
                     var isKeepAlive = notification?.MonitoredItems?.Count == 1 &&
                                       notification?.MonitoredItems?.First().ClientHandle == 0 &&
                                       notification?.MonitoredItems?.First().Message?.NotificationData?.Count == 0;
-                    var sequenceNumber = notification?.MonitoredItems?.First().Message?.SequenceNumber;
+                    var sequenceNumber = (notification?.MonitoredItems?.First()?.Message?.SequenceNumber)
+                                         .GetValueOrDefault(0);
                     var publishTime = (notification?.MonitoredItems?.First().Message?.PublishTime).
                         GetValueOrDefault(DateTime.UtcNow);
 
-                    _logger.Debug("DataChange for subscription '{subscription}'/'{sessionId}', sequence#: "
-                        + "{Sequence} isKeepAlive: {KeepAlive}, publishTime: {PublishTime}",
-                        Id,
-                        Connection.CreateConnectionId(),
-                        sequenceNumber,
-                        isKeepAlive,
-                        publishTime);
+                    if (_currentSequenceNumber == sequenceNumber) {
+                        _logger.Debug("DataChange for subscription '{subscription}'/'{sessionId}', sequence#: "
+                            + "{Sequence} isKeepAlive: {KeepAlive}, publishTime: {PublishTime}",
```

Comment:
Maybe we can have a `,` after `{Sequence}` ?

Reply:
indeed